lib/auth/methods.go

```diff
@@ -382,8 +403,7 @@ func (a *Server) authenticateUserInternal(ctx context.Context, req AuthenticateU
 					Webauthn: wantypes.CredentialAssertionResponseToProto(req.Webauthn),
```

Comment:
Should we tighten the checks above to make sure a nil `req.Pass` won't just slip by? Around line 375, aka `if req.Pass != nil {`.

```go
			if req.Pass == nil && requiredExt.UserVerificationRequirement != string(protocol.UserVerificationRequirement) {
				return nil, trace.BadParameter("password required")
			}
```

Reply:
Nevermind, I see this can't be true for all situations.